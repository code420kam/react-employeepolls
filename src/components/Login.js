import React, { useEffect } from "react";
import AuthUser from "./AuthedUser";
import { useSelector } from "react-redux";
import { useNavigate,useLocation } from "react-router-dom";

const Login = () => {
  const navigate = useNavigate();
  const auth = useSelector((state) => state.login);
  const location = useLocation();
  useEffect(() => {
    if (auth.isAuthenticated) {
      if (location.state !== undefined && 'from' in location.state) {
        navigate(location.state.from);
      } else {
        navigate("/");
      }}
  }, [auth.isAuthenticated]);
  return (
    <React.Fragment>
      <AuthUser />
    </React.Fragment>
  );
};

export default Login;
