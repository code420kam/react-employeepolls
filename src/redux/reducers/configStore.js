import {applyMiddleware } from "redux";
import { legacy_createStore as createStore } from "redux";
import { composeWithDevTools } from 'redux-devtools-extension';
import rootReducer from "./rootReducer";
import thunk from "redux-thunk";

export default function configStore() {
  return createStore(rootReducer, composeWithDevTools(applyMiddleware(thunk)));
}
