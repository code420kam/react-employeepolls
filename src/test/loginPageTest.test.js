import { Provider } from "react-redux";
import Login from "../components/Login";
import AuthUser from "../components/AuthedUser";
import { render} from "@testing-library/react";
import { screen, fireEvent } from "@testing-library/react";
import { Screen } from "@testing-library/react";
import configStore from "../redux/reducers/configStore";
import { BrowserRouter } from "react-router-dom";

describe("Login page testing", () => {
  //snapshot testing
  it("Snapshot test after login created", () => {
    const store = configStore();
    const {loginPage} = render(
      <Provider store={store}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </Provider>
    );
    expect(loginPage).toMatchSnapshot();
  });

  it("Ä°f selectbox value emtpy", () => {
    const store = configStore("");
    const view =render(
      <Provider store={store}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </Provider>
    );
    const cred = screen.getByTestId("selUser");
    fireEvent.change(cred, { target: { value: '' } });
    const submitButton = screen.getByTestId("submitButton");
    expect(submitButton).toBeDisabled();
  });
});