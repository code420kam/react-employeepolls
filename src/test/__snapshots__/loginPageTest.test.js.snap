// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page testing Snapshot test after login created 1`] = `undefined`;
